experimental/examples/bubblebench/editable-shared-tree/src/Bubblebench.ts

```diff
@@ -0,0 +1,110 @@
+        this.root.set("unique-bubblebench-key-1337", this.maybeTree.handle);
+    }
+
+    protected async initializingFromExisting() {
+        // console.log("existing initialization called, delaying tree connection for 3 seconds");
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        this.maybeTree = await this.root
+            .get<IFluidHandle<ISharedTree>>("unique-bubblebench-key-1337")!
+            .get();
+    }
+
+    protected async hasInitialized() {
+        if (this.tree === undefined) {
+            throw new Error(
+                "hasInitialized called but tree is still undefined",
+            );
+        }
+        this.maybeAppState = new AppState(
+            this.tree.root as AppStateTreeProxy,
+            /* stageWidth: */ 640,
+            /* stageHeight: */ 480,
+            /* numBubbles: */ 1,
+        );
+
+        const onConnected = () => {
+            // Out of paranoia, we periodically check to see if your client Id has changed and
+            // update the tree if it has.
+            setInterval(() => {
+                const clientId = this.runtime.clientId;
+                if (
+                    clientId !== undefined &&
+                    clientId !== this.appState.localClient.clientId
+                ) {
+                    this.appState.localClient.clientId = clientId;
+                }
+            }, 1000);
+        };
+
+        // Wait for connection to begin checking client Id.
+        if (this.runtime.connected) {
+            onConnected();
+        } else {
+            this.runtime.once("connected", onConnected);
+        }
+    }
+
+    /**
+     * Initialize the schema of the shared tree to that of the Bubblebench AppState
+     * and inserts a node with the initial AppState as the root of the tree.
+     * @param tree - ISharedTree
+     */
+    initializeTree(tree: ISharedTree) {
+        tree.storedSchema.update(AppStateSchemaData);
+        const initialRootNodeJson = {
+            type: AppStateSchema.name,
+            fields: {
+                clients: [],
+            },
+        };
+        tree.context.root.insertNodes(0, [initialRootNodeJson].map(singleTextCursor));
+    }
+
+    private get tree() {
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        return this.maybeTree!;
+    }
+
+    public get appState() {
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        return this.maybeAppState!;
+    }
+}
+
+/**
+ * The DataObjectFactory declares the Fluid object and defines any additional distributed data structures.
+ * To add a SharedSequence, SharedMap, or any other structure, put it in the array below.
+ */
+export const BubblebenchInstantiationFactory = new DataObjectFactory(
+    Bubblebench.Name,
+    Bubblebench,
+    [new SharedTreeFactory()], // Is this correct?
```

Comment:
Yes, this is fine for now (we will have to adjust this API later to allow control of write format, but for now this is the best you can do)
```suggestion
    [new SharedTreeFactory()],
```

Reply:
Sounds good. I updated this comment to reflect that concern.